### ISSUE TITLE
Webservice customer GET request returns malformed XML

### ISSUE THREAD

**Issue by btjsteens** [PrestaShop/PrestaShop#30895]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

When requesting a customer using the webservice, Prestashop returns XML with illegal tags inside the nested "group" element.
The tag <0> causes parsers to fail.

```
        <associations>
            <groups nodeType="group" api="groups">
                <group xlink:href="https://URL/api/groups/3">
                    <0>
                        <![CDATA[]]>
                    </0>
                </group>
            </groups>
        </associations>

```

### 

### Expected behavior

Should return valid XML

### Steps to reproduce

1. Set up a demo shop
2. Create a webservice key with enough permissions
3. Request https://URL/api/customers/1
4. Parse as XML

### PrestaShop version(s) where the bug happened

8.0.0

**Comment by sallemiines** [comment]
Hello @btjsteens 

Thanx for reporting the issue.

Following your steps , unfortunately I didn't manage to reproduce your isssue 

![image](https://user-images.githubusercontent.com/31663369/212957281-d6d1175f-b549-49d6-8daa-9a7125a7dfd1.png)

Did I miss smthg ? 

PLease check and feedback !

Thank you !

**Comment by btjsteens** [comment]
Hi @sallemiines,

Thanks for your efforts. 
I did a fresh install of the latest Prestashop (8.0.1) in order to reproduce the issue.
Turns out the outcome is different depending on the PHP version:

PHP 7.4:
```
<associations>
            <groups nodeType="group" api="groups">
                <group xlink:href="http://localhost/shop2/api/groups/3">
                    <id>
                        <![CDATA[3]]>
                    </id>
                </group>
            </groups>
        </associations>
```

PHP 8.1:
```
        <associations>
            <groups nodeType="group" api="groups">
                <group xlink:href="http://localhost/shop2/api/groups/3">
                    <0>
                        <![CDATA[]]>
                    </0>
                </group>
            </groups>
        </associations>
```

**Comment by florine2623** [comment]
Hello @btjsteens ,

Your issue is the same as https://github.com/PrestaShop/PrestaShop/issues/30165
It has been fixed in https://github.com/PrestaShop/PrestaShop/pull/30309

The fix will be available in PS 8.1.0.

I close your issue as it is a duplicate !

**Comment by florine2623** [comment]
Duplicate of https://github.com/PrestaShop/PrestaShop/issues/30165
